src/jit/emitxarch.cpp

```diff
@@ -2012,6 +2100,21 @@ inline UNATIVE_OFFSET emitter::emitInsSizeSV(instrDesc* id, code_t code, int var
         prefix += 1;
     }
 
+    // Most 16-bit operand instructions need a prefix
+    if (attrSize == EA_2BYTE && ins != INS_movsx && ins != INS_movzx)
```

Comment:
@CarolEidt, this looks like the cause of the regression. Namely, this was meant to replace the check immediately preceding it here: https://github.com/dotnet/coreclr/pull/24555/files#diff-a51165bc1fd2c0fa3987370e2e714356R2098 (not be in addition to it).

I've fixed this in the latest changes, which I am testing and getting diffs for before pushing.